src/event.h

```diff
@@ -40,6 +41,7 @@ class ev_entry
     int weight;
     bool cbdata;
 
+    CodeContext::Pointer codeContext; ///< event creator's context
```

Comment:
If possible, please use `CodeContextPointer` here so that we can `#include "base/forward.h"` here, in event.h, and `#include "base/CodeContext.h"` in event.cc. Long-term, the whole ev_entry class should probably be moved to event.cc but that move is outside this PR scope.

Reply:
I tried this initially but rejected due to compilation dependencies. Some sources, which create events (and include event.h), require "base/CodeContext.h". We would need to fix all these cc files, adding  "base/CodeContext.h" to them.

Reply:
Strange, I do not see `ev_entry` used in any files other than event.cc, but perhaps I missed some hidden dependency. If you are sure that there are outside `ev_entry` users, please consider this change request addressed.

Reply:
ev_entry is constructed when eventAdd() is called: so the callers of eventAdd() should know about codeContext.

Reply:
No, eventAdd() callers should not know about eventAdd() _implementation_.

Reply:
I initially thought that this may be caused somehow by inlining, but it seems that the reason is different: ev_entry is marked with MEMPROXY_CLASS, which uses sizeof() to calculate the type size, and, hence it should know the size of each member. So, event.h should include "base/CodeContext.h".

Reply:
> MEMPROXY_CLASS, which uses sizeof() to calculate the type size, and, hence it should know the size of each member

Commenting out MEMPROXY_CLASS line still results in a similar build failure.

There is probably some C++ rule that every class declaration has to have a known size, even if the class is unused.

sizeof(CodeContextPointer) becomes known (without knowing what CodeContext is) if we include "base/RefCount.h", but I am OK with the current PR code -- it makes it easier to remember to move the current CodeContext.h `#include` later, when we are going to move ev_entry class declaration to event.cc.